sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -563,6 +507,30 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
         return new LinkSubscription<>(reactorReceiver, subscription);
     }
 
+    /**
+     * Asynchronously waits for the session's active endpoint state.
+     *
+     * @return A mono that completes when the session is active.
+     */
+    private Mono<Void> onActiveEndpoint() {
+        return RetryUtil.withRetry(getEndpointStates().takeUntil(state -> state == AmqpEndpointState.ACTIVE),
```

Comment:
I've been seeing this pattern in a lot of locations, do we want to centralize it anywhere?

Reply:
I don't think this is required to be centralised.